pages/works/[id].tsx

```diff
@@ -0,0 +1,110 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import styles from "@/styles/WorkSinglePage.module.css";
+import { fontSyne, fontSatoshi } from "../_app";
+import works from "@/constants/works.json";
+import Divider from "@/components/base/Divider";
+import Contact from "@/components/general/Contact";
+import InfoItem from "@/components/general/InfoItem";
+
+const WorkSinglePage = () => {
+  const [data, setData] = useState({
+    id: "",
+    title: "",
+    intro: "",
+    info: {
+      client: "",
+      timeline: "",
+      service: "",
+      website: "",
+    },
+    report: {
+      challenge: "",
+      solution: "",
+      result: "",
+    },
+    img: [
+      {
+        src: "",
+        alt: "",
+      },
+      {
+        src: "",
+        alt: "",
+      },
+      {
+        src: "",
+        alt: "",
+      },
+    ],
+  });
```

Comment:
Let's try to define a type and use it in this data state. Initialize this data as the type of the one you have defined or null with this state.

eg:
type User = {
    name: string;
}
const [user, setUser] = useState<User | null>(null);